src/infrastructure/arranging/ordering.rs

```diff
@@ -234,18 +234,25 @@ impl<T> Arranging for OrderingStream<T> {
     /// This can only happen in cases where we have a duplicated package. Again we don't give anything back.
     ///
     /// # Remark
-    /// - When we receive an item there is a possibility that a gab is filled and one or more items will could be returned.
+    /// - When we receive an item there is a possibility that a gap is filled and one or more items will could be returned.
     ///   You should use the `iter_mut` instead for reading the items in order.
     ///   However the item given to `arrange` will be returned directly when it matches the `expected_index`.
     fn arrange(
         &mut self,
         incoming_offset: usize,
         item: Self::ArrangingItem,
     ) -> Option<Self::ArrangingItem> {
+        let incoming_offset = incoming_offset as u16;
         if incoming_offset == self.expected_index {
-            self.expected_index += 1;
+            self.expected_index = self.expected_index.wrapping_add(1);
             Some(item)
-        } else if incoming_offset > self.expected_index {
+        } else if (self.expected_index < u16::max_value() / 2
```

Comment:
this is a bit complex, maybe it will be helpful if there is a small comment saying what the purpose of it is.

Reply:
It is, maybe there's something that exists so we can do `self.expected_index.within_next(u16::max_value(), incoming_offset)`.

Reply:
Should be resolved with the new function